### PR TITLE
[dagster-dlift] get_sources method

### PR DESCRIPTION
**Pull request by dpeng817**
## Summary & Motivation
Method to retrieve dbt sources via rest api.
## How I Tested These Changes
Unit + live.

### CODE REVIEW

**Comment by dpeng817** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25457?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#25458** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25458?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25457** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25457?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#25446** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25446?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25444** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25444?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25443** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25443?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dpeng817 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/experimental/dagster-dlift/dagster_dlift/cloud_instance.py [hunk 1]
```diff
@@ -112,3 +116,26 @@ def get_dbt_models(self, environment_id: int) -> Sequence[Mapping[str, Any]]:
...
+            {"environmentId": environment_id, "first": page_size, "after": start_cursor},
+        ):
+            sources.extend(
+                [
+                    source["node"]
+                    for source in response["data"]["environment"]["definition"]["sources"]["edges"]
+                ]
+            )
+            if not response["data"]["environment"]["definition"]["sources"]["pageInfo"][
+                "hasNextPage"
```
Comment (benpankow) [inline examples/experimental/dagster-dlift/dagster_dlift/cloud_instance.py:135]: seems like this while loop + pagination etc could be pulled out into some shared method, similar to how you've deduplicated code in the mocking test stuff?

**Review by benpankow — APPROVED** [review]

File: examples/experimental/dagster-dlift/dagster_dlift/cloud_instance.py (hunk 1 above)
Comment (dpeng817) [reply examples/experimental/dagster-dlift/dagster_dlift/cloud_instance.py:135]: Yea I buy it; but I also don't want to couple it too finely before we know exactly what we need from all the components

Comment (dpeng817) [reply examples/experimental/dagster-dlift/dagster_dlift/cloud_instance.py]: Ended up finding something that I think feels pretty good.

**Pull request merged**